src/main/scala/gemmini/GemminiConfigs.scala

```diff
@@ -72,6 +73,84 @@ case class GemminiArrayConfig[T <: Data : Arithmetic, U <: Data](
   val mvout_len_bits = log2Up(meshColumns * tileColumns + 1)
   val mvout_rows_bits = log2Up(meshRows * tileRows + 1)
 
+  //==========================================================================
+  // sanity check mesh size
+  //==========================================================================
+  def BLOCK_ROWS = tileRows * meshRows
+  def BLOCK_COLS = tileColumns * meshColumns
+  require(BLOCK_ROWS == BLOCK_COLS, "BLOCK_ROWS != BLOCK_COLS!")
+
+  def DIM             = BLOCK_ROWS
```

Comment:
I'm slightly concerned this will get confused with the DIM in gemmini_params.h.
It might just be my immediate association (since there is not reason there should be an interaction between these DIMs), but just a thought

Reply:
assuming a square mesh, the DIM in GemminiConfig.scala and gemmini_param.h are semantically the same.

Reply:
We're actually hoping to break the requirement of a square mesh. I got started on it earlier, but I ended up failing. But I would like to get it working it eventually.

If we end up permitting non-square matrices, but we don't use any of your CISC commands, would you expect it to cause any problems for code that doesn't use the CISC commands, but which runs on a non-square matrix?

Reply:
correct: if you use a non-square mesh, and never issue a cisc command, the original hardware will not be adversely affected by the new cisc hardware.